app/views/handlers/list_action.py

```diff
@@ -47,8 +47,25 @@ def handle_post(self):
             self.parent_location.block_id
         )
         self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
+        self.evaluate_and_update_section_status()
```

Comment:
This will only update dependencies on a list action of a non-primary person list collector. Even though it is unlikely, we need to update the dependencies on an action of a primary person list action as well, as it is possible to someone to change their mind and add/remove a primary person later on.